src/ocrd_utils/os.py

```diff
@@ -144,37 +167,88 @@ def list_all_resources(executable, moduled=None, xdg_data_home=None):
     processor_path_var = '%s_PATH' % executable.replace('-', '_').upper()
     if processor_path_var in environ:
         for processor_path in environ[processor_path_var].split(':'):
-            if Path(processor_path).is_dir():
-                candidates += Path(processor_path).iterdir()
+            processor_path = Path(processor_path)
+            if processor_path.is_dir():
+                candidates += iterbase(processor_path)
     if 'data' in resource_locations:
         datadir = Path(xdg_data_home, 'ocrd-resources', executable)
         if datadir.is_dir():
-            candidates += datadir.iterdir()
+            candidates += iterbase(datadir)
     if 'system' in resource_locations:
         systemdir = Path(RESOURCES_DIR_SYSTEM, executable)
         if systemdir.is_dir():
-            candidates += systemdir.iterdir()
+            candidates += iterbase(systemdir)
     if 'module' in resource_locations and moduled:
         # recurse fully
-        for resource in itertree(Path(moduled)):
+        base = Path(moduled)
+        for resource in itertree(base):
             if resource.is_dir():
                 continue
             if any(resource.match(pattern) for pattern in
                    # Python distributions do not distinguish between
                    # code and data; `is_resource()` only singles out
                    # files over directories; but we want data files only
                    # todo: more code and cache exclusion patterns!
-                   ['*.py', '*.py[cod]', '*~', 'ocrd-tool.json', 'environment.pickle', 'resource_list.yml', 'lib.bash',
-                    '*.egg-info/*', 'copyright.txt', 'LICENSE*', 'README.md']):
+                   ['*.py', '*.py[cod]', '*~', '.*.swp', '*.swo',
+                    '__pycache__/*', '*.egg-info/*', '*.egg',
+                    'copyright.txt', 'LICENSE*', 'README.md', 'MANIFEST',
+                    'TAGS', '.DS_Store',
+                    # C extensions
+                    '*.so',
+                    # translations
+                    '*.mo', '*.pot',
+                    '*.log', '*.orig', '*.BAK',
+                    '.git/*',
+                    # our stuff
+                    'ocrd-tool.json',
+                    'environment.pickle', 'resource_list.yml', 'lib.bash']):
```

Comment:
Should we move the long list to `ocrd_utils.constants.py`?

Reply:
If it is going to be re-used, yes. But I doubt that...